### ISSUE TITLE
Transaction data inconsistency

### ISSUE THREAD

**Issue by trapp** [ethcore/parity#2169]
I've sent a transaction using sendRawTransaction. It got included in a block and Parity processed that block. However, fetching the transaction using eth_getTransactionByHash still shows it as unconfirmed.

eth_getTransactionByHash (0xf0e70acd0dfa6e3a5906c595d1e7d7a91afb066518b56ebff91bf97384349b5e):
```
{ blockHash: null,
  blockNumber: null,
  creates: null,
  …
  hash: '0xf0e70acd0dfa6e3a5906c595d1e7d7a91afb066518b56ebff91bf97384349b5e',
  … }
```

eth_getBlockByNumber (0x22e7c6, true):
```
{ author: '0xea674fdde714fd979de3edf0f56aa9716b898ec8',
  difficulty: '0x3a7cc569f66f',
  extraData: '0x7573322e65746865726d696e652e6f7267',
  gasLimit: '0x47b760',
  gasUsed: '0x0119c3',
  hash: '0x0766cd28165da54203ee6099f69794e5754c87d695cf3ef3cf85f68bccc229ce',
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000200080000004000000000000000000000000200000000000000000000000000000000000000000000000004000200000000000000000000000000000000100000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000400000000000000000000000000000000',
  miner: '0xea674fdde714fd979de3edf0f56aa9716b898ec8',
  number: '0x22e7c6',
  parentHash: '0xb007466904f71c2aca97a1c84de098140cf0c18b242f8642ec6678c9f37bff03',
  receiptsRoot: '0x834ed04079acb3d4b5a157cc070bf61e3d3bebf8c45d9d6b0d3dd00a77f1add9',
  sealFields:
   [ '0x68bf72e6cf5f876d26e102b2516a356ba4d321bb80408ce85c9a9c25f913fdd0',
     '0x5c62252eb910d4ca' ],
  sha3Uncles: '0xc5d1eae0e1ff8f0fb8fa7599f288e85ac85fbbbb5cce3afa64e79d153fe4e82f',
  size: 1409,
  stateRoot: '0x1a86e85672a11651c003dec2c9768df1a10fecccc75744b643b3fa9e0dbfb3ca',
  timestamp: '0x57dfca8e',
  totalDifficulty: '0x0364da304cb366128e',
  transactions:
   [ …,
     { blockHash: '0x0766cd28165da54203ee6099f69794e5754c87d695cf3ef3cf85f68bccc229ce',
       blockNumber: '0x22e7c6',
       creates: null,
       from: '0xfcde4905379f21f1b25ec1590e34e3de6226db73',
       gas: '0x88b8',
       gasPrice: '0x04a817c800',
       hash: '0xf0e70acd0dfa6e3a5906c595d1e7d7a91afb066518b56ebff91bf97384349b5e',
       input: '0xf7654176',
       nonce: '0x54',
       raw: '0xf871548504a817c8008288b894fa52274dd61e1643d2205169732f29114bc240b3896193def42a2056280084f76541761ca04e54d1ab87e9d0637214e61e9a1dbafe940ec28c26149bbbf64b4f5c8bc10e8ba00e87b5e303d6558b971bb3d9d0dfa0aad9cd9ec2b8a5740f0437273dd08ce6b6',
       to: '0xfa52274dd61e1643d2205169732f29114bc240b3',
       transactionIndex: '0x01',
       value: '0x6193def42a20562800' },
     … ],
  transactionsRoot: '0x4c74aba1b8361bb43c2e67aedb94d4a9adcd38929fe06438dfd20d32f083f6ad',
  uncles: [ '0xb190fb132ce6c85f6b57875bb42186873b20831da0a61abd8d1a89510eb9a6df' ] }
```

This is only the case on the node that broadcasted the transaction, other parity instances return correct data.

**Comment by trapp** [comment]
Note that parity did return it with correct data until it got ~30 confirmations. After that it only shows null as shown above.

**Comment by tomusdrw** [comment]
What Parity version are you running?

**Comment by trapp** [comment]
1.3.0

**Comment by tomusdrw** [comment]
@debris Could you confirm that it's most probably the same re-org bug fixed in `1.3.1` release?

**Comment by debris** [comment]
That's possible

**Comment by trapp** [comment]
I've updated to v1.3.1 and will report again when it happens again.

**Comment by trapp** [comment]
Happens consistently after couple of minutes now with v1.3.1.

eth_getTransactionByHash (0xaa7374443879f29fe852ff0cadb8f72fe53e300857e79ac966359cb0afd20756)
```
{ blockHash: null,
  blockNumber: null,
  creates: null,
  from: '0x7af00838a1ee111f44c2fe5aa99fe951948f1bfd',
  gas: '0x88b8',
  gasPrice: '0x050cc25bd3',
  hash: '0xaa7374443879f29fe852ff0cadb8f72fe53e300857e79ac966359cb0afd20756',
  input: '0xf7654176',
  nonce: '0x24',
  raw: '0xf8702485050cc25bd38288b894fa52274dd61e1643d2205169732f29114bc240b3881bc76b6b6a67305884f76541761ba0aed73f8375a12ddf098e24c2a61d758b8ebf24337c5be7537219dc51b28ce653a0710d217a22e1f03b8efa09714529128aba3e56d5aaec1aedf463ed110acbae60',
  to: '0xfa52274dd61e1643d2205169732f29114bc240b3',
  transactionIndex: null,
  value: '0x1bc76b6b6a673058' }
```

eth_getBlockByNumber (0x22e9c0)
```
{ author: '0xc0ea08a2d404d3172d2add29a45be56da40e2949',
  difficulty: '0x371288c3cabe',
  extraData: '0x7777772e62772e636f6d',
  gasLimit: '0x47b760',
  gasUsed: '0xc80b',
  hash: '0x59a07133ed8a5be7ce4796cff00cbfdf4479f3c6192cceb7d727efb1f25c675f',
  logsBloom: '0x00000004000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000080000004000000000000000000000000000000000000000000000000000000000000000000000000004000200000000000000000000000000000000100000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000400000000000000000000000000000001',
  miner: '0xc0ea08a2d404d3172d2add29a45be56da40e2949',
  number: '0x22e9c0',
  parentHash: '0x9211655f5ea470afb744412a9f23f789e522ea3fc43582518f16da56e4bd3524',
  receiptsRoot: '0x49795cfeb6b626f93d415d6562774c2629201870167d90506243b5735fc34983',
  sealFields:
   [ '0xb274d9cc99a14e88c95762cc4afadce8b0ef2f84f20a5778672b74bd246fc5fb',
     '0x9dcfd42801ce77f7' ],
  sha3Uncles: '0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347',
  size: '0x0309',
  stateRoot: '0x072053d7d7d1d49bc1f128b660bcf181df73a01b4631e4f0c1a2a01a445e7b2c',
  timestamp: '0x57dfeba5',
  totalDifficulty: '0x03654bb69b6040315a',
  transactions:
   [ { blockHash: '0x59a07133ed8a5be7ce4796cff00cbfdf4479f3c6192cceb7d727efb1f25c675f',
       blockNumber: '0x22e9c0',
       creates: null,
       from: '0x7af00838a1ee111f44c2fe5aa99fe951948f1bfd',
       gas: '0x88b8',
       gasPrice: '0x050cc25bd3',
       hash: '0xaa7374443879f29fe852ff0cadb8f72fe53e300857e79ac966359cb0afd20756',
       input: '0xf7654176',
       nonce: '0x24',
       raw: '0xf8702485050cc25bd38288b894fa52274dd61e1643d2205169732f29114bc240b3881bc76b6b6a67305884f76541761ba0aed73f8375a12ddf098e24c2a61d758b8ebf24337c5be7537219dc51b28ce653a0710d217a22e1f03b8efa09714529128aba3e56d5aaec1aedf463ed110acbae60',
       to: '0xfa52274dd61e1643d2205169732f29114bc240b3',
       transactionIndex: '0x00',
       value: '0x1bc76b6b6a673058' },
     … ],
  transactionsRoot: '0x000cae905b1ece37b60233fa1961bd69d66c7a5af39972bf1d055943635a651d',
  uncles: [] }
```

Restarting parity and immediately running the query again fixes it:
```
{ blockHash: '0x59a07133ed8a5be7ce4796cff00cbfdf4479f3c6192cceb7d727efb1f25c675f',
  blockNumber: '0x22e9c0',
  creates: null,
  from: '0x7af00838a1ee111f44c2fe5aa99fe951948f1bfd',
  gas: '0x88b8',
  gasPrice: '0x050cc25bd3',
  hash: '0xaa7374443879f29fe852ff0cadb8f72fe53e300857e79ac966359cb0afd20756',
  input: '0xf7654176',
  nonce: '0x24',
  raw: '0xf8702485050cc25bd38288b894fa52274dd61e1643d2205169732f29114bc240b3881bc76b6b6a67305884f76541761ba0aed73f8375a12ddf098e24c2a61d758b8ebf24337c5be7537219dc51b28ce653a0710d217a22e1f03b8efa09714529128aba3e56d5aaec1aedf463ed110acbae60',
  to: '0xfa52274dd61e1643d2205169732f29114bc240b3',
  transactionIndex: '0x00',
  value: '0x1bc76b6b6a673058' }
```

**Comment by trapp** [comment]
I can confirm this is fixed by #2331 

Thank you!
